### ISSUE TITLE
Axe error in color contrast

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#1185]
Came across this error on the Cauldron website:
https://pattern-library.dequelabs.com/components/fields

Found it using Attest ext, with axe-core 3.1.1

```
TypeError: Cannot read property '0' of undefined at chrome-extension://jhmbiicdkgdffmplcdbmjmifammiaeoj/adapter.bundle.js:1136:225919 at Array.every (<anonymous>) at chrome-extension://jhmbiicdkgdffmplcdbmjmifammiaeoj/adapter.bundle.js:1136:225860 at Array.forEach (<anonymous>) at Object.color.filteredRectStack (chrome-extension://jhmbiicdkgdffmplcdbmjmifammiaeoj/adapter.bundle.js:1136:225740) at Object.color.getBackgroundStack (chrome-extension://jhmbiicdkgdffmplcdbmjmifammiaeoj/adapter.bundle.js:1136:226231) at Object.color.getBackgroundColor (chrome-extension://jhmbiicdkgdffmplcdbmjmifammiaeoj/adapter.bundle.js:1136:226960) at Object.evaluate (chrome-extension://jhmbiicdkgdffmplcdbmjmifammiaeoj/adapter.bundle.js:1136:132087) at Check.../axe-devtools-react/node_modules/axe-core/axe.js.Check.run (chrome-extension://jhmbiicdkgdffmplcdbmjmifammiaeoj/adapter.bundle.js:1109:12265) at chrome-extension://jhmbiicdkgdffmplcdbmjmifammiaeoj/adapter.bundle.js:1109:18224
```

**Comment by marcysutton** [comment]
axe-core's color contrast code is throwing an error when it encounters a DOMRect outside of the viewport, in this case one of the inline code blocks that's cut off by horizontal scrolling. I narrowed it to `getRectStack` method, in particular where it runs `getCoords`.

If an inline element rect is hidden by the viewport, it doesn't return any coordinates. There's no return statement outside of `if (coords)` in the mapping step of `getRectStack`, so that entry in the filtered array ends up `undefined`, cascading down to further code in `getFilteredRectStack` and throwing the error when it compares `rectA` to an undefined `rectB`. 

The question is, what should axe-core return if a DOMRect is offscreen? Maybe the filter step in `getRectStack` should check for valid coordinates before mapping and not just width (that was originally written for line breaks with no width)?

Here's the code: https://github.com/dequelabs/axe-core/blob/develop/lib/commons/color/get-background-color.js#L248

**Comment by aabenoja** [comment]
I am not sure if this is this is related. I get an error thrown when running aXe directly in the browser.

```
"TypeError: Cannot read property '0' of undefined
    at https://unpkg.com/axe-core@3.1.2/axe.js:12483:39
    at Array.every (<anonymous>)
    at https://unpkg.com/axe-core@3.1.2/axe.js:12482:28
    at Array.forEach (<anonymous>)
    at Object.color.filteredRectStack (https://unpkg.com/axe-core@3.1.2/axe.js:12477:21)
    at Object.color.getBackgroundStack (https://unpkg.com/axe-core@3.1.2/axe.js:12497:30)
    at Object.color.getBackgroundColor (https://unpkg.com/axe-core@3.1.2/axe.js:12519:30)
    at Object.evaluate (https://unpkg.com/axe-core@3.1.2/axe.js:8443:29)
    at Check.run (https://unpkg.com/axe-core@3.1.2/axe.js:492:32)
    at https://unpkg.com/axe-core@3.1.2/axe.js:735:15"
```

It looks like it's around [this area](https://github.com/dequelabs/axe-core/blob/89d18d08ed72ba944dfce71c72d5887f4ca26cc1/lib/commons/color/get-background-color.js#L291). I'm not sure if it's because an undefined rectangle is getting pushed to the stack.

**Comment by WilcoFiers** [comment]
Closed by #1341
